### DISCUSSION TITLE
Create extensions when importing data with PostgreSQL

### DISCUSSION THREAD

**Discussion by andwards** [cloudnative-pg/cloudnative-pg#5307] [Q&A]
TLDR I'm trying to bootstrap import a database with postgis functions, but it fails with `pg_restore: error: could not execute query: ERROR:  function postgis.postgis_extensions_upgrade() does not exist`.

I believe this is caused due to the extension not being created yet in the new cluster via `CREATE EXTENSION postgis;`. For `bootstrap.initdb`, it doesn't say what order all the different things execute, but it seems that anything under the `postInit` does not occur until after the database import, not allowing the extension to be created. 

It also appears that all extensions are dropped before restoring, causing a failure due to having extension functions.
https://github.com/cloudnative-pg/cloudnative-pg/blob/v1.23.3/pkg/management/postgres/logicalimport/database.go#L309

```
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: pg-test
spec:
  instances: 3
  imageName: ghcr.io/cloudnative-pg/postgis:15
  storage:
    storageClass: "pg-database"
    size: 150Gi
  bootstrap:
    initdb:
      import:
        type: monolith
        databases:
          - "*"
        source:
          externalCluster: old-cluster
  externalClusters:
    - name: old-cluster
      connectionParameters:
        host: pg-host
        user: pg-user
        dbname: pg-database
      password:
        name: pg-config
        key: pg-password
```
